cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/StrimziPodSetController.java

```diff
@@ -718,4 +721,35 @@ public void onDelete(Pod pod, boolean deletedFinalStateUnknown) {
             enqueuePod(pod, "DELETED");
         }
     }
+
+    private class KafkaEventHandler implements ResourceEventHandler<Kafka> {
+        @Override
+        public void onAdd(Kafka kafka) {
+            // Internationally not used
+        }
+
+        @Override
+        public void onUpdate(Kafka oldKafka, Kafka newKafka) {
+            // Internationally not used
+        }
+
+        @Override
+        public void onDelete(Kafka kafka, boolean deletedFinalStateUnknown) {
+            clearCertificateExpirationMetric(kafka.getMetadata().getName(), kafka.getMetadata().getNamespace());
+        }
+
+        /**
+         * Removes the certificate expiration metric for the Kafka cluster in the namespace specified
+         *
+         * @param clusterName   Name of the Kafka cluster
+         * @param namespace     Namespace of the Kafka cluster
+         */
+        private void clearCertificateExpirationMetric(String clusterName, String namespace) {
+            Tags tags = metrics().getTags(clusterName, namespace, "Kafka");
+            metrics().removeMetric(MetricsHolder.METRICS_CERTIFICATE_EXPIRATION_MS, tags);
+
+            LOGGER.infoOp("Metrics {} for Kafka: {}/{} has been removed",
+                    MetricsHolder.METRICS_CERTIFICATE_EXPIRATION_MS, namespace, clusterName);
+        }
+    }
```

Comment:
This definitely should not be in a StrimziPodSetController.

Reply:
Struggled a lot to find the right place for it, as there are no KafkaController, any suggestion where to put it @scholzj ?

Reply:
Which class exactly do you have in mind @fvaleri or a new one?

Reply:
@steffen-karlsson Honestly, I struggled with this as well. But actually, it seems like something you can place here https://github.com/strimzi/strimzi-kafka-operator/blob/99dbc4393cf93624929713f11ec7a313898122ff/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L818?

Reply:
BTW: You should not need any event handlers there. Just remove the metric I guess.

Reply:
What Jakub suggests should work nicely.

Reply:
Will look into it